cupy/core/core.pyx

```diff
@@ -567,18 +568,37 @@ cdef class ndarray:
         newarray._set_shape_and_strides(shape, strides, False, True)
         return newarray
 
-    def reshape(self, *shape):
+    def reshape(self, *shape, **kwargs):
         """Returns an array of a different shape and the same content.
 
         .. seealso::
            :func:`cupy.reshape` for full documentation,
            :meth:`numpy.ndarray.reshape`
 
         """
-        # TODO(beam2d): Support ordering option
+        cdef int order_char = 'C'
+
         if len(shape) == 1 and cpython.PySequence_Check(shape[0]):
             shape = shape[0]
-        return self._reshape(shape)
+
+        if 'order' in kwargs:
+            order_char = _normalize_order(kwargs['order'], False)
+            if order_char == 'A':
+                if self._f_contiguous:
+                    order_char = 'F'
+                else:
+                    order_char = 'C'
+        if order_char == 'C':
+            return self._reshape(shape)
+        else:
+            # TODO(beam2d): Support ordering option within _reshape instead
+            """
```

Comment:
Let's use line comments (`# `), as we only use this style for docstrings.
Also, please put a blank line before the comment to separate it from the TODO above.